src/ProjectConfiguration.ts

```diff
@@ -617,6 +624,50 @@ export function validateCompanionDefaultWakeInterval(
   return diagnostics;
 }
 
+export function validateTileBuildTarget( tile: Tile, { buildTargets }: AppProjectConfiguration) {
+  if (tile.buildTargets !== undefined){
+    const diagnostic = constrainedSetDiagnostics({
+      actualValues: tile.buildTargets,
+      knownValues: buildTargets,
+      valueTypeNoun: 'tile build targets',
+      notFoundIsFatal: true
+    })
+    return diagnostic;
+  }
+
+  // Return empty list. Nothing to check
+  return new DiagnosticList();
+}
+
+export function validateTileUUID(tile: Tile, config: AppProjectConfiguration) {
+  const diagnostic = new DiagnosticList()
+  if (tile.uuid === undefined) {
+    diagnostic.pushFatalError("Tile UUID can't be undefined");
+  }
+
+  if (!validator.isUUID(String(tile.uuid))) {
+    diagnostic.pushFatalError(
+      'Tile UUID must be a valid UUID',
+    );
+  }
+
+  const knownUUID = [config.appUUID]
+  config.tiles!.forEach((tile) => knownUUID.push(tile.uuid))
+  if (knownUUID.indexOf(tile.uuid) !== knownUUID.lastIndexOf(tile.uuid)) {
+    diagnostic.pushFatalError("Can't have duplicate UUIDs between app and tiles");
+  }
+
+  return diagnostic;
+}
+
+export function validateTileName(tile: Tile, config: AppProjectConfiguration) {
+  const diagnostic = new DiagnosticList()
+  if (tile.name === undefined) {
+    diagnostic.pushFatalError("Tile name can't be undefined");
```

Comment:
nit: flip the error to avoid a double negative `Tile name must be specified`

Reply:
Made the change.